src/main/java/decodes/tsdb/algo/ScalerAdder30.java

```diff
@@ -0,0 +1,290 @@
+	public String input18_MISSING = "ignore";
+	public String input19_MISSING = "ignore";
+	public String input1_MISSING = "ignore";
+	public String input20_MISSING = "ignore";
+	public String input21_MISSING = "ignore";
+	public String input22_MISSING = "ignore";
+	public String input23_MISSING = "ignore";
+	public String input24_MISSING = "ignore";
+	public String input25_MISSING = "ignore";
+	public String input26_MISSING = "ignore";
+	public String input27_MISSING = "ignore";
+	public String input28_MISSING = "ignore";
+	public String input29_MISSING = "ignore";
+	public String input2_MISSING = "ignore";
+	public String input3_MISSING = "ignore";
+    public String input30_MISSING = "ignore";
+	public String input4_MISSING = "ignore";
+	public String input5_MISSING = "ignore";
+	public String input6_MISSING = "ignore";
+	public String input7_MISSING = "ignore";
+	public String input8_MISSING = "ignore";
+	public String input9_MISSING = "ignore";
+	String _propertyNames[] = { "coeff1", 
+    "coeff10", "coeff11", "coeff12", "coeff13", "coeff14", "coeff15", "coeff16", "coeff17", "coeff18", "coeff19", "coeff2", 
+    "coeff20", "coeff20", "coeff21", "coeff22", "coeff23", "coeff24", "coeff25", "coeff26", "coeff27", "coeff28", "coeff29",
+    "coeff30","coeff3", "coeff4", "coeff5", "coeff6", "coeff7", "coeff8", "coeff9", 
+    "input10_MISSING", "input11_MISSING", "input12_MISSING", "input13_MISSING", "input14_MISSING", "input15_MISSING", "input16_MISSING", "input17_MISSING", "input18_MISSING", "input19_MISSING", "input1_MISSING", 
+    "input20_MISSING", "input21_MISSING", "input22_MISSING", "input23_MISSING", "input24_MISSING", "input25_MISSING", "input26_MISSING", "input27_MISSING", "input28_MISSING", "input29_MISSING",
+    "input2_MISSING", "input30_MISSING", "input3_MISSING", "input4_MISSING", "input5_MISSING", "input6_MISSING", "input7_MISSING", "input8_MISSING", "input9_MISSING" };
+//AW:PROPERTIES_END
+
+	// Allow javac to generate a no-args constructor.
+
+	/**
+	 * Algorithm-specific initialization provided by the subclass.
+	 */
+	protected void initAWAlgorithm( )
+		throws DbCompException
+	{
+//AW:INIT
+		_awAlgoType = AWAlgoType.TIME_SLICE;
+//AW:INIT_END
+
+//AW:USERINIT
+//AW:USERINIT_END
+	}
+	
+	/**
+	 * This method is called once before iterating all time slices.
+	 */
+	protected void beforeTimeSlices()
+		throws DbCompException
+	{
+//AW:BEFORE_TIMESLICES
+//AW:BEFORE_TIMESLICES_END
+	}
+
+	/**
+	 * Do the algorithm for a single time slice.
+	 * AW will fill in user-supplied code here.
+	 * Base class will set inputs prior to calling this method.
+	 * User code should call one of the setOutput methods for a time-slice
+	 * output variable.
+	 *
+	 * @throws DbCompException (or subclass thereof) if execution of this
+	 *        algorithm is to be aborted.
+	 */
+	protected void doAWTimeSlice()
+		throws DbCompException
+	{
+//AW:TIMESLICE
+		double tally = 0.0;
+		if (!isMissing(input1))
```

Comment:
This comment is for our future selves not work to do right now... there has got to be a better way to handle this.... so much repeated logic, but using java reflection is probably too slow. Might be worth an investigation in the future